src/main/kotlin/com/exactpro/th2/kafka/client/KafkaConnection.kt

```diff
@@ -0,0 +1,188 @@
+        val kafkaRecord = ProducerRecord<String, ByteArray>(topic, value)
+        producer.send(kafkaRecord) { _, exception: Throwable? ->
+            when (exception) {
+                null -> {
+                    val msgText = "Message '${message.metadata.id}' sent to Kafka"
+                    LOGGER.info(msgText)
+                    eventSender.onEvent(msgText, "Send message", message)
+                }
+                else -> {
+                    val msgText = "Failed to send message '${message.metadata.id}' to Kafka"
+                    LOGGER.error(msgText, exception)
+                    eventSender.onEvent(msgText, "SendError", message, exception)
+                }
+            }
+        }
+    }
+
+    override fun run() {
+        try {
+            consumer.subscribe(config.topicToAlias.keys)
+            while (!Thread.currentThread().isInterrupted) {
+                val records = consumer.poll(Duration.ofMillis(DURATION_IN_MILLIS))
+
+                if (!records.isEmpty) {
+                    val inactivityPeriod = System.currentTimeMillis() - records.first().timestamp()
+                    if (inactivityPeriod > config.maxInactivityPeriodMillis) {
+                        consumer.seekToEnd(getAllPartitions())
+                        val msgText = "Inactivity period exceeded ($inactivityPeriod ms). Skipping unread messages."
+                        LOGGER.info { msgText }
+                        eventSender.onEvent(msgText, "ConnectivityServiceEvent")
+                    } else {
+                        for (record in records) {
+                            val messageID = factory.newMessageIDBuilder()
+                                .setConnectionId(
+                                    ConnectionID.newBuilder()
+                                        .setSessionAlias(config.topicToAlias[record.topic()])
+                                        .setSessionGroup(config.sessionGroup)
+                                        .build()
+                                )
+                                .setDirection(Direction.FIRST)
+                                .setSequence(firstSequence())
+                                .setTimestamp(Instant.now().toTimestamp())
+                                .build()
+                            messageProcessor.process(
+                                RawMessage.newBuilder()
+                                    .setMetadata(RawMessageMetadata.newBuilder().setId(messageID))
+                                    .setBody(ByteString.copyFrom(record.value()))
+                                    .build()
+                            )
+                        }
+                    }
+
+                    consumer.commitAsync { offsets: Map<TopicPartition, OffsetAndMetadata>, exception: Exception? ->
+                        if (exception != null) {
+                            LOGGER.error(exception) { "Commit failed for offsets $offsets" }
+                        }
```

Comment:
Please, add also trace login in case we successfully committed the current position